sdk/src/main/java/io/honeycomb/opentelemetry/HoneycombSdk.java

```diff
@@ -246,10 +247,19 @@ public HoneycombSdk buildAndRegisterGlobal() {
          * @see GlobalOpenTelemetry
          */
         public HoneycombSdk build() {
-            Preconditions.checkNotNull(apiKey, "apiKey must be non-null");
-            Preconditions.checkNotNull(dataset, "dataset must be non-null");
             Preconditions.checkNotNull(sampler, "sampler must be non-null");
 
+            Logger logger = Logger.getLogger(HoneycombSdk.class.getName());
+
+            if (apiKey == null) {
```

Comment:
Same as above, I think we should check for null and empty strings here too.

Reply:
There's definitely a lot we still _can_ do to validate things up front (we don't check that the URL is well-formed, for example). Added the empty string check, that was easy enough to add onto this.